src/Generator.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Generator.hpp"
+
+Lines getAllVectorPermutation_(int vSize)
+{
+    std::vector<int> row(vSize);
+    Lines result;
```

Comment:
```cpp
Lines result(row.size(), row);
```

This should spare line 10, vector will be constructed with the content here.

Reply:
Line 10 is called multiple time to save every new permutation found.

Reply:
Yes it is modified. At beginning of the loop we know that row is valid and unique. At end of loop we may reach end of all possible permutation (iterator i will be less then 0, so next row[i] would be invalid).